### DISCUSSION TITLE
Inaccurate thermochemistry for species ionically bonded with potassium

### DISCUSSION THREAD

**Discussion by CMDePompa** [ReactionMechanismGenerator/RMG-Py#2710] [Q&A]
Hi! I am trying to get thermochemistry for some species that are ionically bonded to potassium. For now, I am comparing calculations for KCN, KOH, and K2SO4 with their experimental data in JANAF tables to verify potassium is accurately treated in the calculations. Unfortunately, the Arkane calculated enthalpies are nowhere near correct, and consequently, neither are the Gibbs free energy calculations. The Cp and S values are pretty close to experimental data. Might anyone have any insight into this problem?

**Comment by JacksonBurns** [comment]
I think @rwest can probably speak to this the best, but my understanding is that RMG doesn't handle ions (see [here](https://github.com/ReactionMechanismGenerator/RMG-Py/issues/2659#issuecomment-2100672408)), so perhaps Arkane is in the same boat?

**Reply by CMDePompa** [reply]
Thanks for getting back to me. I made a change to /rmg/RMG-database/input/quantum_corrections/data.py at line 48, the `atom_hf` value for potassium atom to match the value from [JANAF](https://janaf.nist.gov/tables/K-005.html). I also added a library index for potassium atom to the main thermo library at /rmg/RMG-database/input/thermo/libraries/primaryThermoLibrary.py:

```
entry(
    index=47,
    label="K",
    molecule="""
multiplicity 2
1 K u1 p0 c0
""",
    thermo=ThermoData(
        Tdata=([0, 100, 200, 250, 298.15, 300, 350, 400, 450, 500, 600, 700, 800, 900, 1000, 1039.54, 1100, 1200, 1300, 1400, 1500, 1600, 1700, 1800, 1900, 2000, 2100, 2200, 2300, 2400, 2500, 2600, 2700, 2800, 2900, 3000, 3100, 3200, 3300, 3400, 3500, 3600, 3700, 3800, 3900, 4000, 4100, 4200, 4300, 4400, 4500, 4600, 4700, 4800, 4900, 5000, 5100, 5200, 5300, 5400, 5500, 5600, 5700, 5800, 5900, 6000
], "K"),
        Cpdata=([0, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96797323135755, 4.96821223709369, 4.96869024856597, 4.96964627151052, 4.97155831739962, 4.97466539196941, 4.97992351816444, 4.98757170172084, 4.99856596558317, 5.01314531548757, 5.03250478011472, 5.05664435946463, 5.0862810707457, 5.12213193116635, 5.16395793499044, 5.21247609942639, 5.2684034416826, 5.33150095602294, 5.40272466539197, 5.48183556405354, 5.57026768642447, 5.66778202676864, 5.76553537284895, 5.87930210325048, 6.00071701720841, 6.13432122370937, 6.27868068833652, 6.434034416826, 6.6008604206501, 6.77461759082218, 6.96080305927342, 7.16108986615679, 7.37284894837476, 7.44383365200765, 7.64890057361377, 7.86137667304015, 8.07050669216061, 8.23709369024857, 8.45721797323136, 8.68044933078394, 8.9060707456979, 9.13288718929254, 9.36018164435946, 9.58675908221797, 9.81166347992352, 10.0334608030593, 10.1522466539197, 10.3577915869981, 10.557361376673, 10.75
], "cal/(mol*K)"),
        H298=(21.27, "kcal/mol"),
        S298=(38.322, "cal/(mol*K)"),
    ),
    shortDesc="""library value for reference potassium""",
    longDesc="""

""",
)
```

My advisor and I decided we needed the atom specifically due to the nature of the enthalpy calculations. There is [JANAF ](https://janaf.nist.gov/tables/K-006.html) data for K+ in the gas phase. I will try putting that in as well to see if maybe that is the source of error.

Thanks again for the help!
